fortran/plumed_f08.f90

```diff
@@ -450,6 +459,20 @@ subroutine plumed_f_f08_eh(error_ptr,code,what_ptr,opt_ptr) bind(C)
            error%what(j:j)=p_chars(j)
          enddo
        endif
+       if (C_associated(opt_ptr)) then
+         call C_F_pointer(opt_ptr,opt, [huge(0)])
+         do i = 1, huge(0),2
+           if (.not. c_associated(opt(i))) exit
+           if (c_associated(opt(i+1))) then
+             call C_F_pointer(opt(i),opt_key)
+             if (opt_key == "n") then
+               call C_F_pointer(opt(i+1),error_nested)
+               allocate(error%nested)
+               error_nested=c_loc(error%nested)
```

Comment:
Thanks @aradi!

The way it currently works is:
- `plumed_f_f08_eh` is called passing an optional `opt_ptr`. In C this is a `void**` pointer, pointing to an array of pointers with an even number of slots, null terminated
- for every pair of slots, I have (a) a pointer to a letter and (b) a generic pointer. The pointer to a letter says which is information is passed. PLUMED should not assume that the wrapper is using the passed information, to maintain backward compatibility.
- In plumed 2.8 I am only using one letter ("c") to pass the error code for some specific C++ exception type
- Here I introduce a new letter ("n") to pass a pointer to a location. If the wrapper can react to a nested exception, it populates the location pointed with a pointer to another allocated exception
- PLUMED checks if the location was populated. If not, it does not pass the nested exception. If yes, it calls again `plumed_f_f08_eh` using that pointer as an argument

I don't plan to pass multiple "n" codes from PLUMED, so this will be executed only once. And setting `error_nested` means setting the location pointed to by `error_nested`, so it is used later by PLUMED.

I was by the way thinking if this could be extended to Python 3.11 ExceptionGroup, and I would do it with a separate code (e.g. "g"), which could then be repeatable multiple times. But I did't implement anything like that yet.

Reply:
I see. I overlooked the `pointer` attribute of `error_nested`, with that it makes sense, indeed. I'd suggest to add an `exit` before line 472. Then the intent, that you only look for the first occurrence of, is much more explicit.